### PR TITLE
OpenCV3 migration

### PR DESCRIPTION
**Pull request by bbferka**
- modifications such that RS can be run on both indigo and kinetic without any additional effort
- biggest changes occurred in KNNClassifier and FeatureExtractor;

### CODE REVIEW

File: descriptors/analysis_engines/demo.xml [hunk 1]
```diff
@@ -141,14 +141,14 @@
...
        <node>NormalEstimator</node>
        <node>PlaneAnnotator</node>
        <node>ImageSegmentationAnnotator</node>
        <node>PointCloudClusterExtractor</node>
        <node>ClusterMerger</node>
-       <node>ResultAdvertiser</node>
+       <node>ResultAdvertiser</node>-->
       </fixedFlow>
     </flowConstraints>
     <typePriorities/>
```
Comment (jworch) [inline descriptors/analysis_engines/demo.xml]: Changes to demo.xml should not be committed. Will break the tutorial.

File: src/recognition/include/rs/recognition/LinemodInterface.h [hunk 2]
```diff
@@ -52,8 +56,8 @@ class LinemodInterface
   /*
    * Processes a frame
    */
-  void process(const cv::Mat &color, const cv::Mat &depth, std::vector<Result> &results, const float minResponse = 85.0f,
-               const std::vector<std::string> &classes = std::vector<std::string>(), const cv::Mat &mask = cv::Mat());
+  void process(const cv::Mat &color, const cv::Mat &depth, std::vector<Result> &results, float minResponse = 85.0f,
```
Comment (jworch) [inline src/recognition/include/rs/recognition/LinemodInterface.h]: Why did you remove the 'const'?

File: descriptors/analysis_engines/demo.xml (hunk 1 above)
Comment (bbferka) [reply descriptors/analysis_engines/demo.xml]: yea, you're right:) I'll uncomment;

File: src/recognition/include/rs/recognition/LinemodInterface.h (hunk 2 above)
Comment (bbferka) [reply src/recognition/include/rs/recognition/LinemodInterface.h]: no real reason. mistake; will fix

File: src/utils/include/rs/utils/DepthImageProcessing.h [hunk 3]
```diff
@@ -24,8 +24,8 @@
 #include <pcl/point_types.h>
 #include <opencv2/opencv.hpp>
 #include <image_geometry/pinhole_camera_model.h>
-#include <opencv2/gpu/gpu.hpp>
-#include <opencv2/gpu/gpumat.hpp>
+//#include <opencv2/gpu/gpu.hpp>
+//#include <opencv2/gpu/gpumat.hpp>
```
Comment (bbferka) [inline src/utils/include/rs/utils/DepthImageProcessing.h]: these includes I will delete
